### PR TITLE
DAOS-16127 tools: Add daos health check command (#14730)

### PR DESCRIPTION
**Pull request by mjmac**
Perform basic system health checks from the client
perspective. Checks the following:

  * Client/Server versions
  * Key library versions and paths
  * Connected sytem information
  * Pool status for all pools to which the user
    has access
  * Container status for all containers in the
    checked pools

Signed-off-by: Michael MacDonald <mjmac@google.com>

### CODE REVIEW

File: src/client/api/SConscript
Comment (daltonbohning) [inline src/client/api/SConscript:1]: Are the builds changes here what introduced the ubuntu build failure in master?
https://github.com/daos-stack/daos/actions/runs/10274977022/job/28432755992

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

File: src/client/api/SConscript
Comment (mjmac) [reply src/client/api/SConscript:1]: Ah, right. Totally forgot about that issue. Is there a ticket for it?

Comment (daltonbohning) [reply src/client/api/SConscript:1]: I don't think so :( I just remembered it now. I'll go create a ticket...

Comment (daltonbohning) [reply src/client/api/SConscript:1]: https://daosio.atlassian.net/browse/DAOS-16313

Comment (daltonbohning) [reply src/client/api/SConscript:1]: Anyway, it's just a landing build, so I think if the fix can come quickly enough we might not need to hold up this PR?

**Review by tanabarr — DISMISSED** [review]
Please add clean/unclean cherry pick label. Thanks

File: src/client/api/SConscript
Comment (mjmac) [reply src/client/api/SConscript:1]: I'm testing a fix... I think it works? https://github.com/daos-stack/daos/actions/runs/10286752033/job/28468199032?pr=14889

Comment (daltonbohning) [reply src/client/api/SConscript:1]: Looks like it :)

Comment (mjmac) [reply src/client/api/SConscript:1]: OK, I went ahead and cherry-picked that fix onto this. It has to run the full `Features: control pool` gauntlet anyhow, so not much time lost.

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM. Thanks!

**Comment by mjmac** [comment]
Test failure appears to be an instance of [DAOS-16035](https://daosio.atlassian.net/browse/DAOS-16035). The other failures are the usual UCX/MD-on-SSD failures that happen when I forget to deselect those stages. :/

**Comment by mjmac** [comment]
@tanabarr: Mind giving this a +1 so we can get it landed? TIA

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**